lib/Team/CollectiveTeamResourceProvider.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Collectives\Team;
+
+use OCA\Collectives\AppInfo\Application;
+use OCA\Collectives\Db\CollectiveMapper;
+use OCA\Collectives\Service\CollectiveService;
+use OCP\IURLGenerator;
+use OCP\Teams\ITeamResourceProvider;
+use OCP\Teams\TeamResource;
+
+class CollectiveTeamResourceProvider implements ITeamResourceProvider {
+	public function __construct(private CollectiveMapper $collectiveMapper, private CollectiveService $collectiveService, private IURLGenerator $urlGenerator, private ?string $userId) {
+
+	}
+
+	public function getId(): string {
+		return Application::APP_NAME;
+	}
+
+	public function getName(): string {
+		return 'Collective';
+	}
+
+	public function getIconSvg(): string {
+		return file_get_contents(__DIR__ . '/../../img/collectives.svg');
+	}
+
+	public function getSharedWith(string $teamId): array {
+		$collective = $this->collectiveMapper->findByCircleId($teamId);
+		if ($collective) {
+			$collective = $this->collectiveService->getCollectiveInfo($collective->getId(), $this->userId);
```

Comment:
This might throw `NotFoundException` if the user doesn't have access to the collective. Should we handle this gracefully, or it it ok to throw? When I read the server code correctly, at least here it would be a problem if one provider throws:

https://github.com/nextcloud/server/blob/7033967115ae7f5e5be8551f0530143d5b1a7885/lib/private/Teams/TeamManager.php#L80C18-L89C1